tests/baselines/reference/spreadTypeVariable.errors.txt

```diff
@@ -17,13 +17,13 @@ spreadTypeVariable.ts(14,12): error TS2698: Spread types may only be created fro
     function f3<T extends number | string[]>(arg: T) {
       return { ...arg }
                ~~~~~~
-!!! error TS2698: Spread types may only be created from object types.
+!!! error TS2698: Spread appears unintentional because this value is not possibly an object.

```

Comment:
Need better logic here - `T` is constrained to `string[]` which is an object

Reply:
@RyanCavanaugh

Reply:
@geryogam This is likely going to be quite difficult, and unfortunately I can't allocate time to help new contributors with a "You can send a PR if you want to"-tier issue.

Reply:
Okay, since you think it's going to be quite difficult and you're the development lead for the TypeScript team at Microsoft, I don't think I'll even try to go any further, as I don't have much time to devote to open source contributions at the moment. Thanks for the information.